oeps/oep-0017-feature-toggles.rst

```diff
@@ -0,0 +1,809 @@
+  enabling for everyone.
+
+Once the team is confident about their change and the change is released to all users,
+they would safely remove the gating toggle.
+
+.. _Mean Time to Recovery (MTTR): https://www.thoughtworks.com/radar/techniques/focus-on-mean-time-to-recovery
+.. _canary release: https://martinfowler.com/bliki/CanaryRelease.html
+
+Use Case 4: Ops - Graceful Degradation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _Ops - Graceful Degradation: `Use Case 4: Ops - Graceful Degradation`_
+
+In certain cases, the development team (in consultation with the operations team)
+may choose to extend the lifetime of an Ops toggle in the codebase even after
+releasing its gated feature. A small number of such long-lived Ops "kill switches"
+provide operators dynamic controls to gracefully degrade the system under high load.
+Operators can use these circuit-breaker capabilities either preemptively in the
+anticipation of a high-demand event or in response to taming an unanticipated high
+load or attack.
+
+Typically, long-lived Ops toggles are useful for gating non-critical features that
+are very expensive on system resources. However, the long-term costs of maintaining
+the added complexity in the code should be measured against the benefits of
+operationally degrading the service when needed.
+
+Use Case 5: Beta Testing
+~~~~~~~~~~~~~~~~~~~~~~~~
+.. _Beta Testing: `Use Case 5: Beta Testing`_
+
+For user-facing changes, the engineering and product teams may choose to release them
+to a *specific subset* of the population before releasing to the rest. This is in
+contrast to the `Ops - Monitored Rollout`_ case where changes are rolled out to a
+*random subset* of users.
+
+In the edX case, the Beta testing program may include the following types of population
+subsets:
+
+* Users - list of specific users
+
+* Courses - users enrolled in any course within a list of specific courses *(for 
+  course-related features)*
+
+* Content-provider Organizations - users enrolled in any course offered by any
+  organization in a list of specific organizations *(for course-related features)*
+
+* User-provider Organizations - enterprise users associated with any organization in
+  a list of specific organizations
+
+The feature toggle is useful during the duration of the Beta testing period and is 
+removed afterward.
+
+Long-term Business
+~~~~~~~~~~~~~~~~~~
+
+There are sometimes business requirements for keeping long-term feature toggles in
+order to expose or limit certain features to certain groups.
+
+Use Case 6: VIP / White Label
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _`VIP / White Label`: `Use Case 6: VIP / White Label`_
+
+The business may choose to modify the product experience for different classes of
+users. For example, the state of a feature toggle may depend on whether the user
+is a paying customer, a bundled feature within a subscription plan, or applicable
```

Comment:
This sounds like entitlements to me. In light of not having a robust entitlements system baked into our platform, I can see why someone would try to leverage toggles as a short term fix but l would caution against using toggles for this and not sure it belongs in an OEP.

Ideally an entitlement system would be tied back to reporting systems, financial systems, etc. and I have concern if we lean too heavy on long-term feature toggles it may make building a robust system more difficult.